sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc

```diff
@@ -0,0 +1,790 @@
+
+_T_ is restricted to types that have a trivial destructor and a trivial default constructor in this revision of the specification (the constructor restriction may be partially relaxed in a future revision). _propertyListT_ enables properties to be associated with a `device_global`.
+
+Since _T_ is restricted to types with trivial default constructors in this version of the specification, there are no non-default `device_global` constructors, and therefore no initialization values may be specified for the content of a `device_global` allocation on a device.
+
+The allocation of type _T_ is zero-initialized on each device prior to the first access to that `device_global` variable.
+
+`device_global` may only be declared with static storage duration at namespace scope or class scope. If a `device_global` is declared with any other storage duration or scope, the program is ill-formed.
+
+The example below creates two global namespace scope `device_global` objects named `dm1` and `dm2`. `dm1` contains one object of type `MyClass` on each device, and the `device_global` object has external linkage.  `dm2` contains an array of four integers on each device, and the `device_global` object has internal linkage.  In both cases, the `MyClass` and `int[4]` allocations on each device are zero-initialized before any non-initialization accesses occur.
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi;
+
+device_global<MyClass> dm1;
+static device_global<int[4]> dm2;
+
+int main () {
+  sycl::queue Q;
+  Q.submit([&](sycl::handler& h) {
+    h.single_task([=]() {
+      int x = 5;
+      if (dm1.get().flag)
+        x = dm2[0];
+    });
+  });
+}
+----
+
+Properties may be specified for a `device_global` to provide semantic modification or optimization hint information to the compiler.  Specific properties are defined in other extensions, but example uses of a property (with a "no copy" attribute described by another extension) are:
+
+[source,c++]
+----
+device_global<MyClass, property_list_t<no_copy::value_t>> dm1;
+device_global<int[4], property_list_t<no_copy::value_t>> dm2;
+----
+
+[NOTE]
+====
+
+On a device, `device_global` has similar semantics to a reference wrapper.  The dot operator (`operator.`) cannot be overloaded, so a `get()` member is provided to allow a reference to be extracted directly when needed.  Some operators are declared in `device_global` that must be members (e.g. `operator[]` and `+operator->+`).  Note that other operators can be overloaded by specific `T` as free functions, which will be selected through implicit conversion to `T` in device functions.
+
+====
+
+
+The section below and the table following describe the constructors, member functions and factory methods for `device_global`.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+
+template <typename T, typename property_listT = property_list<>>
```

Comment:
Nit: Using upper camel case (i.e. `PropertyListT`) for template arguments would help avoid confusion with existing typenames, and makes it more obvious when something else in the class definition is referring back to a template parameter.  I think it's especially important here, because the similarly named `property_list_t` exists in the `SYCL_EXT_ONEAPI_PROPERTY_LIST` extension.

Reply:
Fixed.